### PR TITLE
RPRBLND-1770: Blender Material Library - Online Integration

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
Online server for the MaterialX material Library has been prepared. Thus we will make so the user does not install the material library at all.
When opening the plugin, a list of existing materials (and thumbnails) will be downloaded and cached from the server.  
When adding a material to the scene, download the material from the library and textures locally.

### EFFECT OF CHANGE
Made basic implementation of online MaterialX library (hided by default from user):
- Added Material Library panel in materials tab
- Added possibility to see materials thumbnails and import material into MaterialX nodetree.

### TECHNICAL STEPS
1. Created utils/matlib.py with dataclasses for getting materials info from server. Implemented basic caching of downloaded files.
2. Added Material Library UI panel into material tab:
    - Created properties classes WindowManagerProperties, MatlibProperties.
    - Panel class HDUSD_MATLIB_PT_matlib with list of available materials with thumbnails.
    - Operator HDUSD_MATLIB_OP_import_material to import selected material.
3. Adjusted MxNodeTree.import_() for importing materials from matlib:
    - added prepare_for_import() which adjusts mx.Document with \<surfaceshader> node instead \<shaderref> statement
    - added parsing of internal \<nodedef> statement inside imported .mtlx file.
4. Added algib MxNode classes, which are used a lot in matlib materials:
    - added deps/mx_libs with alglib nodedefs
    - added alglib/alglib_defs.mtlx to generated node classes.
5. Added config.engine_use_preview for quick enable/disable preview mode for development purposes.
6. Added matlib_enabled setting in config.py to hide matlib panel from user.

### NOTES FOR REVIEWERS
This is initial implementation. There are still several things to do: UI improvement (more thumbnails, options and info about material), improve materials caching, show downloading progress, etc.

### CODE REVIEW

**Review by VascoPi — APPROVED** [review]
Tested, works fine.

File: src/hdusd/config.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 logging.limit_log('', level_show_min=logging.INFO)
 
+matlib_url = "https://matlibapi.cistest.luxoft.com/api"
+engine_use_preview = True
```
Comment (DagerD) [inline src/hdusd/config.py:20]: Should these vars be capitalized (since they are const)?

**Review by DagerD — APPROVED** [review]
Tested. Works fine.

File: src/hdusd/config.py (hunk 1 above)
Comment (bnagirniak) [reply src/hdusd/config.py:20]: not here

**Pull request merged**